activerecord/test/cases/calculations_test.rb

```diff
@@ -358,6 +358,12 @@ def test_count_with_distinct
     assert_equal 4, Account.select(:credit_limit).uniq.count
   end
 
+  def test_count_with_distinct_emits_single_distinct_in_query
+    assert_equal 4, Account.select(:credit_limit).distinct.count
+    queries = assert_sql { Account.select(:credit_limit).distinct.count}
+    assert_equal 1, queries.last.scan(/DISTINCT/).count
```

Comment:
Result sets are not different, because additional distinct doesn't make a difference - it is a distinct out of one value - count, which is the same value.